src/Microsoft.Health.Dicom.Blob/Registration/Experimental/IBlobServiceBuilder.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public interface IBlobServiceBuilder
```

Comment:
I'm thinking if the `IxxBuilder` is a good pattern.
* Pros:
One pros I see is it prevents the long extension list for `IServiceCollection`
![image](https://user-images.githubusercontent.com/67345894/123123415-aa290880-d3fb-11eb-9809-30a18c2e4c8d.png)

* Cons:
1. Extensibility: 
The first extension method which returns `IxxBuilder` has to be called, if you don't want it, you have to create your own `IxxBuilder` implemention (e.g: this duplicate [code](https://github.com/microsoft/dicom-server/blob/03a281fe62a7a46bf7498ccc80ce10501e2ec723/src/Microsoft.Health.Dicom.Functions/Registration/ServiceCollectionExtensions.cs#L44)), which I'm really not a favor.

Actually the project we need is `Microsoft.Health.Dicom.Metadata` instead of `Microsoft.Health.Dicom.Blob`, in this case, how should we make this builder pattern extended to that project?

* Propose
 Not like  [IFunctionsHostBuilder](https://github.com/Azure/azure-functions-dotnet-extensions/blob/main/src/Extensions/DependencyInjection/IFunctionsHostBuilder.cs), whose implementation is [here](https://github.com/Azure/azure-functions-dotnet-extensions/blob/main/src/Extensions/DependencyInjection/FunctionsHostBuilder.cs), our builder is a simply wrapper on `IServiceCollection`. 
So maybe we could get rid of this builder pattern.